test/harness/asyncHelpers-asyncTest-func-throws-sync.js

```diff
@@ -0,0 +1,16 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+description: |
+    The 'asyncTest' helper reports synchronous errors via $DONE.
+includes: [asyncHelpers.js]
+---*/
+var called = false;
+function $DONE(error) {
+  called = true;
+  assert(error instanceof Test262Error);
+}
+asyncTest(function () {
+  throw new Test262Error("Should not be rethrown");
```

Comment:
Maybe we should make this a string or something like that, so it's easy to assert that it's the exact same thrown value that is passed to `$DONE`?